sklearn/cluster/_hdbscan/hdbscan.py

```diff
@@ -646,7 +648,7 @@ class HDBSCAN(ClusterMixin, BaseEstimator):
             None,
             Interval(Integral, left=1, right=None, closed="left"),
         ],
-        "metric": [StrOptions(FAST_METRICS | {"precomputed"}), callable],
+        "metric": [StrOptions(FAST_METRICS | {"precomputed", "cosine"}), callable],
```

Comment:
To also include other valid pairwise metrics:
```suggestion
        "metric": [StrOptions(FAST_METRICS | set(_VALID_METRICS) | {"precomputed"}), callable],
```
`_VALID_METRICS` can be imported from `metrics.pairwise`

Reply:
I agree, it's not just about cosine. We need to allow all metrics accepted by `pairwise_distances`